src/accelerate/accelerator.py

```diff
@@ -2289,12 +2295,16 @@ def gather_for_metrics(self, input_data):
                         "The used dataset had no length, returning gathered tensors. You should drop the remainder yourself."
                     )
                     return data
-                elif self.gradient_state.remainder > 0:
+                elif self.gradient_state.remainder > 0 and self.distributed_type != DistributedType.NO:
```

Comment:
Nice I think this winds up solving another issue I saw somewhere about samples being dropped